health-check/README.md

```diff
@@ -0,0 +1,179 @@
+# Health Checks
+
+Suppose we have a simple Couper proxy configuration – requests to `/` are forwarded to a backend service:
+
+```hcl
+server {
+  api {
+    endpoint "/" {
+      proxy {
+        backend = "fragile_backend"
+      }
+    }
+  }
+}
+
+definitions {
+  backend "fragile_backend" {
+    origin = "http://backend:8080"
+  }
+}
+```
+
+To demonstrate Couper's health monitoring facilities, we create an unreliable
+backend service that is sometimes failing. To be exact, it actually fails half of the time:
+
+```hcl
+server {
+  api {
+    endpoint "/" {
+      response {
+        # fail every 15 seconds for a period of 15 seconds
+        status = (unixtime() % 30) < 15 ? 200 : 500
+      }
+    }
+  }
+}
+```
+
+Let's start the services to see them in action:
+
+```sh
+$ docker-compose pull && docker-compose up
+
+```
+
+Then, we send a request…
+
+```sh
+$ curl -i localhost:8080/
+HTTP/1.1 200 OK
+Date: Mon, 13 Jun 2022 09:21:35 GMT
+…
+```
+
+… and Couper responds with `200 OK`. A few seconds later the backend is failing and
+accordingly Couper forwards the `500 Internal Server Error`:
+
+```sh
+$ curl -i localhost:8080/
+HTTP/1.1 500 Internal Server Error
+Date: Mon, 13 Jun 2022 09:21:47 GMT
+…
+```
+
+Now let's monitor the healthiness of the backend with the `beta_health` block:
+
+```hcl
+definitions {
+  backend "fragile_backend" {
+    origin = "http://backend:8080"
+    beta_health {}
+  }
+}
+```
+
+Couper will now send `GET` requests to http://backend:8080/ every second and check
+the response status. By default, status codes of 200, 204 or 301 will be considered as healthy.
+If you want to accept other the these codes, you can set the `expected_status` attribute accordingly.
+To change the check frequency, set the `interval` attribute, for example
+
+```hcl
+    beta_health {
+      expected_status = [200, 418]
+      interval = "3s"
+    }
+```
+
+Let's restart the services and filter the logs to better see what's going on:
```

Comment:
Since the watch option is enabled, Couper will handle this change.

Reply:
See 96d9e8390bc7bb42fdade9195c6ba01dc015d801